adoc/extensions/sycl_khr_group_interface.adoc

```diff
@@ -0,0 +1,479 @@
+class work_group {
+
+ public:
+  using id_type = id<Dimensions>;
+  using linear_id_type = size_t;
+  using range_type = range<Dimensions>;
+#if defined(__cpp_lib_mdspan)
+  using extents_type = std::dextents<size_t, Dimensions>;
+#endif
+  using size_type = size_t;
+  static constexpr int dimensions = Dimensions;
+  static constexpr memory_scope fence_scope = memory_scope::work_group;
+
+  work_group(group<Dimensions> g) noexcept;
+
+  operator group<Dimensions>() const noexcept;
+
+  /* -- common by-value interface members -- */
+
+  /* -- common group interface members -- */
+
+};
+
+} // namespace khr
+} // namespace sycl
+----
+
+.[apidef]#work_group constructor#
+[source,role=synopsis,id=api:khr-group-interface-work-group-constructor]
+----
+work_group(group<Dimensions> g) noexcept;
+----
+
+_Effects_: Constructs a [code]#work_group# representing the same collection of
+work-items as [code]#g#.
+
+'''
+
+.[apidef]#work_group conversion operator#
+[source,role=synopsis,id=api:khr-group-interface-work-group-conversion-operator]
+----
+operator group<Dimensions>() const noexcept;
+----
+
+_Returns_: A [code]#group# representing the same collection of work-items as
+this [code]#work_group#.
+
+'''
+
+[[sec:khr-group-interface-sub_group]]
+== [code]#sub_group# class
+
+The [code]#sub_group# class template encapsulates all functionality required to
+represent a particular <<sub-group>> within a kernel.
+It is not user-constructible.
+
+The SYCL [code]#sub_group# class template provides common by-value semantics
+(see <<sec:byval-semantics>>) and the common group interface (see
+<<sec:khr-group-interface-common>>).
+
+[source,role=synopsis]
+----
+namespace sycl {
+namespace khr {
+
+class sub_group {
+
+ public:
+  using id_type = id<1>;
+  using linear_id_type = uint32_t;
+  using range_type = range<1>;
+#if defined(__cpp_lib_mdspan)
+  using extents_type = std::dextents<uint32_t, 1>;
+#endif
+  using size_type = uint32_t;
+  static constexpr int dimensions = 1;
+  static constexpr memory_scope fence_scope = memory_scope::sub_group;
+
+  sub_group(sycl::sub_group sg) noexcept;
+
+  operator sycl::sub_group() const noexcept;
+
+  constexpr size_type max_size() const noexcept;
+
+  /* -- common by-value interface members -- */
+
+  /* -- common group interface members -- */
+
+};
+
+} // namespace khr
+} // namespace sycl
+----
+
+.[apidef]#sub_group constructor#
+[source,role=synopsis,id=api:khr-group-interface-sub-group-constructor]
+----
+sub_group(sycl::sub_group sg) noexcept;
+----
+
+_Effects_: Constructs a [code]#sub_group# representing the same collection of
+work-items as [code]#sg#.
+
+'''
+
+.[apidef]#sub_group conversion operator#
+[source,role=synopsis,id=api:khr-group-interface-sub-group-conversion-operator]
+----
+operator sycl::sub_group() const noexcept;
+----
+
+_Returns_: A [code]#sycl::sub_group# representing the same collection of
+work-items as this [code]#sub_group#.
+
+'''
+
+.[apidef]#max_size#
```

Comment:
```suggestion
.[apidef]#khr::sub_group::max_size#
```

Use the fully qualified name here (without `sycl`).  Otherwise, there will be a conflict with other identifiers named `max_size` in other classes.

Same for the other `[apidef]` in this KHR.

Reply:
Fixed as part of de356628056c3d1034fcde2e2be59f9c4ebf3ecc.